feat(connectors): move to Connector SDK 0.2 (#1255)
camunda/camunda-platform-docs: docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md

takes care of that. Every environment can define its own way of providing such a secret store. Consult
the [runtime environments](#runtime-environments) section to learn more about them.

To replace secrets in the `connectorRequest` using the API, you need to annotate the object's
attributes that can hold secrets with `@Secret`. This instructs the SDK to search for and replace

> secrets in the annotated properties. You can use the `@Secret` annotation on primitive Java types

Author: ```suggestion
secrets in the annotated properties. You can use the `@Secret` annotation on String fields
```

Reviewer: Even better than my suggestion. Clear statement that secrets can only be contained in strings.

and container types. Using the annotation on container objects will lead to
graph traversal until either no further matching annotation is found or the annotated property

> is a primitive Java type.

Reviewer: The wording "Primitive Java Type" might be misleading. "Primitive type" refers to the JSON primitive types Text, Number and Boolean. Text is represented by a CharSequence which is usually implemented by String.

Maybe the more precise wording could be:

... or the annotated property is a String. Marking Numbers or Boolean fields as `@Secret`will result in a runtime exception!

Author: Yeah I like that 👍

> is a primitive Java type.

Author: ```suggestion
is of type String. Annotating properties that neither of type String nor of a container type will lead to runtime exceptions.
```

> is of type String. Annotating properties that are neither of type String nor of a container type will lead to runtime exceptions.

Author: ```suggestion
is of type String. Annotating properties that are neither a String nor a container will lead to runtime exceptions.
```
